src/test/java/com/google/devtools/build/lib/analysis/RunfilesRepoMappingManifestTest.java

```diff
@@ -0,0 +1,274 @@
+import com.google.devtools.build.lib.bazel.bzlmod.FakeRegistry;
+import com.google.devtools.build.lib.bazel.bzlmod.ModuleFileFunction;
+import com.google.devtools.build.lib.bazel.repository.RepositoryOptions.CheckDirectDepsMode;
+import com.google.devtools.build.lib.skyframe.PrecomputedValue;
+import com.google.devtools.build.lib.skyframe.PrecomputedValue.Injected;
+import com.google.devtools.build.lib.skyframe.SkyFunctions;
+import com.google.devtools.build.lib.vfs.Path;
+import com.google.devtools.build.skyframe.SkyFunction;
+import com.google.devtools.build.skyframe.SkyFunctionName;
+import java.io.IOException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests that the repo mapping manifest file is properly generated for runfiles. */
+@RunWith(JUnit4.class)
+public class RunfilesRepoMappingManifestTest extends BuildViewTestCase {
+  private Path moduleRoot;
+  private FakeRegistry registry;
+
+  @Override
+  protected ImmutableList<Injected> extraPrecomputedValues() {
+    try {
+      moduleRoot = scratch.dir("modules");
+    } catch (IOException e) {
+      throw new IllegalStateException(e);
+    }
+    registry = FakeRegistry.DEFAULT_FACTORY.newFakeRegistry(moduleRoot.getPathString());
+    return ImmutableList.of(
+        PrecomputedValue.injected(
+            ModuleFileFunction.REGISTRIES, ImmutableList.of(registry.getUrl())),
+        PrecomputedValue.injected(ModuleFileFunction.IGNORE_DEV_DEPS, false),
+        PrecomputedValue.injected(ModuleFileFunction.MODULE_OVERRIDES, ImmutableMap.of()),
+        PrecomputedValue.injected(
+            BazelModuleResolutionFunction.CHECK_DIRECT_DEPENDENCIES, CheckDirectDepsMode.WARNING));
+  }
+
+  @Override
+  protected AnalysisMock getAnalysisMock() {
+    // Make sure we don't have built-in modules affecting the dependency graph.
+    return new AnalysisMock.Delegate(super.getAnalysisMock()) {
+      @Override
+      public ImmutableMap<SkyFunctionName, SkyFunction> getSkyFunctions(
+          BlazeDirectories directories) {
+        return ImmutableMap.<SkyFunctionName, SkyFunction>builder()
+            .putAll(
+                Maps.filterKeys(
+                    super.getSkyFunctions(directories),
+                    fnName -> !fnName.equals(SkyFunctions.MODULE_FILE)))
+            .put(
+                SkyFunctions.MODULE_FILE,
+                new ModuleFileFunction(
+                    FakeRegistry.DEFAULT_FACTORY, directories.getWorkspace(), ImmutableMap.of()))
+            .buildOrThrow();
+      }
+    };
+  }
+
+  /**
+   * Sets up a Bazel module simple_rule@1.0, which provides a simple_binary rule that passes along
+   * runfiles in the data attribute. It also uses a toolchain (type //:toolchain_type, default
+   * toolchain //:simple_toolchain; the macro //:defs.bzl%simple_toolchain can be used to create
+   * more).
+   */
+  @Before
+  public void setupSimpleBinaryRule() throws Exception {
```

Comment:
mind adding test cases for:
1. dependencies caused by aspects
2. host/exec dependencies
3. dependencies caused by an late-bound attribute 

Based on how the machinery that collects the set of transitive packages works, it looks like all these should work, but it's better to test it.

Reply:
I'll get to this next. re "host/exec dependencies" though -- why would we want those in runfiles at all?

Reply:
Files from host/exec dependencies should never be in the runfiles tree, but to the best of my knowledge, this is not enforced anywhere, which means that it's pretty certain that someone does it anyway.

Therefore, packages of host/exec dependencies must also be in the runfiles repo mapping (unless you convince me that the above property *is* enforced, but I'd be very surprised)

Reply:
ah I see. I just added a new filtering so that only repos contributing runfiles would appear in the runfiles manifest (https://github.com/bazelbuild/bazel/pull/16321#discussion_r977272663), so I think we should have this case covered (unless someone is actively including stuff from host/exec deps into their runfiles, in which case they can simply... not).